pkg/decoder/decoder.go

```diff
@@ -23,27 +25,47 @@ func NewPayload(content []byte, offset int64) *Payload {
 	return &Payload{content, offset}
 }
 
-// Decoder splits raw data based on `\n`, and sends those messages to a channel
+// Decoder splits raw data into messages using '\n' for single-line logs and re for multi-line logs
+// and sends them to a channel
 type Decoder struct {
 	InputChan  chan *Payload
 	OutputChan chan message.Message
+	countDownDuration time.Duration
+	lineBuffer *bytes.Buffer
 	msgBuffer  *bytes.Buffer
+	re         *regexp.Regexp
+	timer      *time.Timer
+	singleline bool
```

Comment:
nitpick: in many places, we do 
```
if d.singleline {
    return
}
```
I'd go instead with `isMultiLine bool`, and 
```
if d.isMultiLine {
    ...
}
```
I find this easier to read